lib/kafka_rest_client.js

```diff
@@ -100,16 +107,36 @@ KafkaRestClient.prototype.getProduceInterval = function getProduceInterval(retry
 KafkaRestClient.prototype.scheduleTopicMapRefresh = function scheduleTopicMapRefresh() {
     var self = this;
     if (self.proxyRefreshTime > 0) {
-        // console.log('Trying to schedule topic refresh job every: ' +
-        // self.proxyRefreshTime / 1000 + ' seconds.');
         /* eslint-disable no-undef,block-scoped-var */
         self.refreshIntervalId = setInterval(function refreshTopics() {
+            totalWaitingTimeMs = 0;
+            self.scheduleConnect(0);
+        }, self.proxyRefreshTime);
+        /* eslint-enable no-undef,block-scoped-var */
+    }
+};
+
+// Reconnect to rest proxy if scheduleTopicMapRefresh fails
+KafkaRestClient.prototype.scheduleConnect = function scheduleConnect(retry) {
+    var self = this;
+    if (retry < connectWaitTimeMs.length) {
+        if (self.proxyRefreshTime > 0 && totalWaitingTimeMs + connectWaitTimeMs[retry] >= self.proxyRefreshTime) {
+            return;
+        }
+
+        self.connecting = true;
```

Comment:
@cloudXane If we set connecting to true here, we need to limit the retry times as I did in line 223, since we have a really long reconnecting periode; but I think it is also fine to set connecting to false, all possible loss messages are only in the period defaultReconnectRetryMs*maxRetries;